sdk/trace/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java

```diff
@@ -251,7 +251,7 @@ Clock getClock() {
 
   @Override
   public <T> ReadWriteSpan setAttribute(AttributeKey<T> key, T value) {
-    if (key == null || key.getKey() == null || key.getKey().length() == 0 || value == null) {
```

Comment:
Removing the unnecessary key null check (null key mapped to empty string now) is originally what had me find the other ones we never removed